treeherder/perf/secretary_tool.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from datetime import (datetime,
+                      timedelta)
+
+import simplejson as json
+from django.conf import settings as django_settings
+
+from treeherder.perf.models import (BackfillRecord,
+                                    BackfillReport,
+                                    PerformanceSettings)
+
+logger = logging.getLogger(__name__)
+
+
+class SecretaryTool:
+    """
+    Tool used for doing the secretary work in the Performance Sheriff Bot.
+    """
+    TIME_TO_MATURE = timedelta(hours=4)
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def validate_settings(cls):
+        perf_sheriff_settings, created = PerformanceSettings.objects.get_or_create(
+            name="perf_sheriff_bot",
+            defaults={"settings": cls._get_default_settings()},
+        )
+
+        if created:
+            logger.info("Performance settings for perf_sheriff_bot not found. Creating with defaults.")
+            return
+
+        # reset limits if the settings expired
+        settings = json.loads(perf_sheriff_settings.settings)
+        if cls.are_expired(settings):
+            logger.info(f"Settings are expired. Expired settings: {settings}")
+
+            perf_sheriff_settings.settings = cls._get_default_settings()
+            perf_sheriff_settings.save()
+
+    @classmethod
+    def mark_reports_for_backfill(cls):
+        # get the backfill reports that are mature, but not frozen
+        mature_date_limit = datetime.utcnow() - cls.TIME_TO_MATURE
+        mature_reports = BackfillReport.objects.filter(frozen=False, last_updated__lte=mature_date_limit)
```

Comment:
Would it help to have an index on `frozen` and `last_updated`to make this query faster?  Or is the table going to be small enough that it won't make a difference?

Reply:
there will be ~1000 items in the table, I don't think it needs an index. @ionutgoldan do you think there can be a case where the number of reports gets high enough to need an index here?

Reply:
We will expire summaries & reports older than a year. In general, the amount of summaries *equals* that of reports.
For the last year, we had a bit more than 3,500 summaries. So that'd be the current size of the reports table.

There are cases when this size can increase. If we enable the **platforms_microbench** framework, that yearly size will increase to 4,000.
Also, if we increase number of pageload perf tests, this size will increase even more.

I don't think this table will have more than 10,000 rows.
For the next 1.5 years, I'm pretty sure we won't even reach 6,000 rows.

Thus, I don't think it's worth adding an index for a 10K-rows table.